definitions/npm/react-router-dom_v6.x.x/flow_v0.201.x-/react-router-dom_v6.x.x.js

```diff
@@ -226,6 +226,246 @@ declare module 'react-router-dom' {
 
   declare export type InitialEntry = string | Partial<Location>;
 
+  declare type LowerCaseFormMethod =
+    | 'get'
+    | 'post'
+    | 'put'
+    | 'patch'
+    | 'delete';
+  declare type UpperCaseFormMethod =
+    | 'GET'
+    | 'POST'
+    | 'PUT'
+    | 'PATCH'
+    | 'DELETE';
+
+  /**
+   * Users can specify either lowercase or uppercase form methods on <Form>,
+   * useSubmit(), <fetcher.Form>, etc.
+   */
+  declare export type HTMLFormMethod =
+    | LowerCaseFormMethod
+    | UpperCaseFormMethod;
+
+  declare type V7_FormMethod = UpperCaseFormMethod;
+
+  declare export type FormEncType =
+    | 'application/x-www-form-urlencoded'
+    | 'multipart/form-data';
+
+  /**
+   * Potential states for state.navigation
+   */
+  declare export type NavigationStates = {
+    Idle: {
+      state: 'idle',
+      location: void,
+      formMethod: void,
+      formAction: void,
+      formEncType: void,
+      formData: void,
+    },
+    Loading: {
+      state: 'loading',
+      location: Location,
+      formMethod: RemixFormMethod | V7_FormMethod | void,
+      formAction: string | void,
+      formEncType: FormEncType | void,
+      formData: FormData | void,
+    },
+    Submitting: {
+      state: 'submitting',
+      location: Location,
+      formMethod: RemixFormMethod | V7_FormMethod,
+      formAction: string,
+      formEncType: FormEncType,
+      formData: FormData,
+    },
+  };
+
+  declare export type Navigation = NavigationStates[$Keys<NavigationStates>];
+
+  declare export type RevalidationState = 'idle' | 'loading';
+
+  /**
+   * Potential states for fetchers
+   */
+  declare type FetcherStates<TData = any> = {
+    Idle: {
+      state: 'idle',
+      formMethod: void,
+      formAction: void,
+      formEncType: void,
+      formData: void,
+      data: TData | void,
+      ' _hasFetcherDoneAnything '?: boolean,
```

Comment:
Is this meant to have the spaces around it? A question for this line and the repeated ones below
```suggestion
      '_hasFetcherDoneAnything'?: boolean,
```